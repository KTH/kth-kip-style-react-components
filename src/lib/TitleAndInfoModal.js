import React from 'react'
import InfoModalButton from './InfoModalButton'

const styles = {
  span: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start'
  },
  btnInfoModal: {
    // /specific for this project
    verticalAlign: 'middle'
  },
  h3: {
    marginTop: '15px',
    marginBottom: '0'
  }
}
const TitleAndInfoModal = ({ modalId, titleAndInfo, btnClose, children }) => {
  const { header, body } = titleAndInfo
  const infoModalLabels = {
    header,
    body,
    btnClose: btnClose || 'Close'
  }
  return (
    <span className='title-and-info' style={styles.span}>
      <h3 data-testid={modalId + '-heading'} style={styles.h3}>
        {header}
        <InfoModalButton
          style={styles.btnInfoModal}
          modalId={modalId + '-infoModal'}
          modalLabels={infoModalLabels}
        />
      </h3>
      {children}
    </span>
  )
}

export default TitleAndInfoModal
