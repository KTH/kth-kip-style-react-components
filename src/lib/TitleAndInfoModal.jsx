import React from 'react'
import { InfoModalButton } from '@kth/kth-kip-style-react-components'

const TitleAndInfoModal = ({ modalId, titleInfo, btnClose }) => {
  const { header, body } = titleInfo
  const infoModalLabels = {
    header,
    body,
    btnClose: btnClose || 'Close'
  }
  return (
    <span className='title-and-info'>
      <h3 data-testid={modalId + '-heading'}>
        {header}
        <InfoModalButton
          modalId={modalId + '-infoModal'}
          modalLabels={infoModalLabels}
        />
      </h3>
    </span>
  )
}

export default TitleAndInfoModal
