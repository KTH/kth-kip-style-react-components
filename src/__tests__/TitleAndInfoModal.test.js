import React from 'react'
import renderer from 'react-test-renderer'
import TitleAndInfoModal from '../lib/TitleAndInfoModal'

const TEST_TRANSLATIONS = {
  header: 'Course content',
  body: '<p>These help insturctions will be displayed when user click on modal icon \'info\'</p>'
}
const BUTTONS_TRANSLATIONS = {
  btnClose: 'Close'
}

describe('TitleAndInfoModal', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<TitleAndInfoModal modalId='courseContent' titleAndInfo={TEST_TRANSLATIONS} btnClose={BUTTONS_TRANSLATIONS.btnClose} />
      ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
