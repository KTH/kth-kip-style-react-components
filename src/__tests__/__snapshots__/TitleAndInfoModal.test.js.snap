// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleAndInfoModal renders properly 1`] = `
<span
  class="title-and-info"
>
  <h3
    data-testid="courseContent-heading"
  >
    Course content
    <span>
      <button
        type="button"
        class="btn-info-modal btn btn-secondary"
        aria-label="Info">
      </button>
    </span>
  </h3>
</span>
`;
